Makefile

```diff
@@ -50,6 +50,10 @@ LDFLAGS += -X helm.sh/helm/v3/internal/version.metadata=${VERSION_METADATA}
 LDFLAGS += -X helm.sh/helm/v3/internal/version.gitCommit=${GIT_COMMIT}
 LDFLAGS += -X helm.sh/helm/v3/internal/version.gitTreeState=${GIT_DIRTY}
 
+# Define constants for the client-go version
+LDFLAGS += -X helm.sh/helm/v3/pkg/lint/rules.k8sVersionMajor=$(shell awk '/client-go/{s=$$2;sub("^v","",s);split(s,v,".");print v[1]+1}' go.mod)
+LDFLAGS += -X helm.sh/helm/v3/pkg/lint/rules.k8sVersionMinor=$(shell awk '/client-go/{s=$$2;sub("^v","",s);split(s,v,".");print v[2]}' go.mod)
```

Comment:
Ah! This is clever! I like it!

Reply:
Can we do the same for [capabilities,go](https://github.com/helm/helm/blob/6898ad14576463ea1df857bb17f2f0ee47653756/pkg/chartutil/capabilities.go#L34-L36) too? It would be wonderful to no longer need to update these variables.

Reply:
done(ish). There's still a bunch of fixed-strings in yaml that is used for validating results. I'm hesitant to do anything about that since the goal is to have a fixed target for the tests.